enacts/calc.py

```diff
@@ -804,9 +860,65 @@ def seasonal_cess_date(
         grouped_daily_data[soil_moisture.name]
         .groupby(grouped_daily_data["seasons_starts"])
         .map(
-            cess_date,
+            cess_date_from_sm,
+            dry_thresh=dry_thresh,
+            dry_spell_length_thresh=dry_spell_length_thresh,
+        )
+    ).rename("cess_delta")
+    # Get the seasons ends
+    seasons_ends = grouped_daily_data["seasons_ends"].rename({"group": time_dim})
+    seasonal_cess_date = xr.merge([seasonal_data, seasons_ends])
+
+    # Tip to get dates from timedelta search_start_day
+    # seasonal_onset_date = seasonal_onset_date[time_dim]
+    # + seasonal_onset_date.onset_delta
+    return seasonal_cess_date
+
+
+def seasonal_cess_date_from_rain(
```

Comment:
This function is mostly identical to seasonal_cess_date_from_sm. If we decide to keep the functions separate, I would like to factor out the common parts. I see how to do it, but I won't bother writing it down until we reach a decision on splitting or merging the functions.

Reply:
This was mostly to have the Maprooms work. If our work in PR #372 turn out successfully, those named seasonal_ xxx functions will become obsolete and we will apply our seasonal operators to whatever functions.

Reply:
OK. Do you want to merge this now and revisit later, or should we keep it open until PR #372 is finalized?

Reply:
I'd rather merge because we aren't sure how longer PR #372 will take; because from here, I also have to see what are the next steps in making the "main" water balance function step-wise; and because this PR also rearranges the tests and the functions locations and I'd rather not have that type of change hanging any longer.